keyboards/sofle/keymaps/default/keymap.c

```diff
@@ -0,0 +1,369 @@
+    _M_QWERTY,
+    _W_QWERTY,
+    _M_COLEMAK,
+    _W_COLEMAK,
+    _M_LOWER,
+    _W_LOWER,
+    _M_RAISE,
+    _W_RAISE,
+    _ADJUST,
+};
+
+enum custom_keycodes {
+    KC_M_QWERTY = SAFE_RANGE,
+    KC_W_QWERTY,
+    KC_M_COLEMAK,
+    KC_W_COLEMAK,
+    KC_M_LOWER,
+    KC_W_LOWER,
+    KC_M_RAISE,
+    KC_W_RAISE,
+    KC_ADJUST,
+};
+
+/* Linux/win variants */
+#define KC_W_UNDO    LCTL(KC_Z)
+#define KC_W_CUT     LCTL(KC_X)
+#define KC_W_COPY    LCTL(KC_C)
+#define KC_W_PASTE   LCTL(KC_V)
+#define KC_W_PRVWD   LCTL(KC_LEFT)
+#define KC_W_NXTWD   LCTL(KC_RIGHT)
+#define KC_W_LSTRT   KC_HOME
+#define KC_W_LEND    KC_END
+#define KC_W_DLINE   LCTL(KC_BSPC)
+
+/* Mac variants */
+#define KC_M_UNDO  LGUI(KC_Z)
+#define KC_M_CUT   LGUI(KC_X)
+#define KC_M_COPY  LGUI(KC_C)
+#define KC_M_PASTE LGUI(KC_V)
+#define KC_M_PRVWD LALT(KC_LEFT)
+#define KC_M_NXTWD LALT(KC_RIGHT)
+#define KC_M_LSTRT LGUI(KC_LEFT)
+#define KC_M_LEND  LGUI(KC_RIGHT)
+#define KC_M_DLINE LGUI(KC_BSPC)
+
+#define KC_____ KC_TRNS
+#define KC_XXXX KC_NO
+#define KC_RESET RESET
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+/*
+ * QWERTY
+ * ,-----------------------------------------.                    ,-----------------------------------------.
+ * |  `   |   1  |   2  |   3  |   4  |   5  |                    |   6  |   7  |   8  |   9  |   0  |  `   |
+ * |------+------+------+------+------+------|                    |------+------+------+------+------+------|
+ * | ESC  |   Q  |   W  |   E  |   R  |   T  |                    |   Y  |   U  |   I  |   O  |   P  | Bspc |
+ * |------+------+------+------+------+------|                    |------+------+------+------+------+------|
+ * | Tab  |   A  |   S  |   D  |   F  |   G  |-------.    ,-------|   H  |   J  |   K  |   L  |   ;  |  '   |
+ * |------+------+------+------+------+------|       |    |       |------+------+------+------+------+------|
+ * |LShift|   Z  |   X  |   C  |   V  |   B  |-------|    |-------|   N  |   M  |   ,  |   .  |   /  |RShift|
+ * `-----------------------------------------/       /     \      \-----------------------------------------'
+ *            | LGUI | LAlt | LCTR |LOWER | /Enter  /       \Space \  |RAISE | RCTR | RAlt | RGUI |
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+[_M_QWERTY] = LAYOUT_kc( \
+  GRV,   1,   2,    3,    4,    5,                     6,    7,    8,    9,    0,  GRV, \
+  ESC,   Q,   W,    E,    R,    T,                     Y,    U,    I,    O,    P,  BSPC, \
+  TAB,   A,   S,    D,    F,    G,                     H,    J,    K,    L, SCLN,  QUOT, \
+  LSFT,  Z,   X,    C,    V,    B, MUTE,       XXXX,   N,    M, COMM,  DOT, SLSH,  RSFT, \
+           LCTRL,LALT,LGUI,M_LOWER, ENT,       SPC,  M_RAISE, RGUI, RALT, RCTRL \
+),
+[_W_QWERTY] = LAYOUT_kc( \
```

Comment:
You don't need a second layer here, for this.  You can use bootmagic to handle this. 
https://docs.qmk.fm/#/feature_bootmagic?id=keycodes

Specifically the `MAGIC_*_CTL_GUI` keycodes will swap the Control and GUI keycodes, rather than needing a separate layer.

Reply:
I have already considered converting this to MAGIC_*_CTL_GUI, but there would be still few keys not covered by that (next/prev word etc). So I would prefer to keep it as it is until I figure out how to deal with those.

If it's an issue which would prevent this PR from being accepted, I think I could remove those shenanigans from the default layout so it's more basic  and move them to an alternative layout (this is in the end based on my personal layout).

Reply:
The media keys should work on both systems, as long as you use the non-OS specific variants.  Eg, these: 
https://docs.qmk.fm/#/keycodes_basic?id=media-keys

I can verify this, as I switch between macOS and Windows on a regular bases, and use Ubuntu on a laptop. 

However, you could just use the one set of keycodes, and use `process_record_user` to override the behavior based on the `keymap_config.lgui_lctl_swap` boolean.  


>If it's an issue which would prevent this PR from being accepted, I think I could remove those shenanigans from the default layout so it's more basic and move them to an alternative layout (this is in the end based on my personal layout).

No, it won't block it, but you're essentially adding a lot of redundancy here, and taking up a lot more firmware space, for relatively little gain, IMO.  And I'd rather point you to a solution that may be better, than just accept the code.  

However, if you really would rather keep it as is, that's okay with me.

Reply:
Oh, I see. I think the `keymap_config.lgui_lctl_swap` was the missing piece I needed. Thanks a lot. You are right, it's better to change it.

Reply:
@drashna So I removed the unnecessary layers, and handled few remaining keys in process_record_user. Thanks for all the suggestions.